src/main/java/net/sf/jabref/gui/PreviewPanel.java

```diff
@@ -306,9 +307,19 @@ public void update() {
         );
         String newValue = sb.toString();
 
-        previewPane.setText(newValue);
-        previewPane.revalidate();
-
+        if (SwingUtilities.isEventDispatchThread()) {
```

Comment:
Could this kind of code (checking for isEventDispatch etc.) extracted to some util method getting passed a function?